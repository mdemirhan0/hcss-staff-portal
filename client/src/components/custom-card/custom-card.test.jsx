import React from 'react';
import { render, screen } from '@testing-library/react';
import CustomCard from './custom-card.component';
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import userEvent from '@testing-library/user-event';

describe('Tests for CustomCard', () => {
    test('Custom Card matches the snapshot', () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <CustomCard
                    imageUrl="#"
                    title="Test"
                    children={null}
                    to="/test"
                    buttonText="Button Text"
                    header="View Test"
                />
            </Router>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should go to the correct url when card is clicked', () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <CustomCard
                    imageUrl="#"
                    title="Test"
                    children={null}
                    to="/test"
                    buttonText="Button Text"
                    header="View Test"
                />
            </Router>
        );

        expect(screen.queryByText(/button text/i)).toBeInTheDocument();

        screen.getByText(/button text/i).click();
        expect(history.location.pathname).toBe("/test");
    });
});
