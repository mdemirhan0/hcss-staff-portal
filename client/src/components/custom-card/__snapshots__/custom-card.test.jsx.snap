// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CustomCard render Custom Card 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-2"
            style="background-image: url(#);"
            title="Test"
          />
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              View Test
            </h2>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardActions-root MuiCardActions-spacing"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            href="/test"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Button Text
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-2"
          style="background-image: url(#);"
          title="Test"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            View Test
          </h2>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
          href="/test"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Button Text
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
